gymnasium/studio-asset-generator
# overhaul everything

## What this PR will do:

- Generate (partial) code snippets for the **course outline** content in the Open edX Studio repos using a single content source for _both_ the `about/overview.html` and the hideous JSON mess in `info/updates.items.json`
- Make it realllly easy to copy/paste the snippets

## Usage/Testing Notes

- Pull this PR/branch
- Build locally
- The index page lists the three courses that are currently set up for testing
- The course asset pages use prism.js for displaying the code snippets and each of the snippets has a copy button to grab the snippet (if it's not visible, click anywhere in the styled code blocks and it'll appear on the right-hand side)

### Warnings/Caveats:

- It is unclear if HTML encoded entities will work in the JSON garbage — **this needs to be tested**
- The _only_ way to make this magic work is with the liquid logic etc present in each of the course pages... i.e. it is not at all DRY but it won't work as an include or in a layout
- The course outline content itself is in a YAML data file... it could be moved into the front matter of each of the .md pages but that might limit our reuse of the data... so it should remain separate (for now at least)
- Don't use smart quotes/apostrophes in the course outline content... use dumb ones and let the markdown parser do it
- GYM-109 has inline quotations in one of the lesson descriptions that were converted to HTML entities (like the other fancy punctuation)... you have been warned (ahem... JSON)

#### What this branch should have been called: 

_Exciting_ New Direction

https://www.youtube.com/watch?v=Z0pmNDIXymc
